### PR TITLE
openssl: 1.0.2a

### PR DESCRIPTION
**Pull request by vszakats**

### CODE REVIEW

**Comment by vszakats** [comment]
I've updated the mirror link to @DomT4. The file is not there yet at this moment, but I'm sure it will be there very soon.

**Comment by DomT4** [comment]
https://github.com/Homebrew/homebrew/pull/37862#issuecomment-83592987

File: Library/Formula/openssl.rb [hunk 1]
```diff
@@ -1,9 +1,9 @@
 class Openssl < Formula
   homepage "https://openssl.org"
-  url "https://www.openssl.org/source/openssl-1.0.2.tar.gz"
-  mirror "https://raw.githubusercontent.com/DomT4/LibreMirror/master/OpenSSL/openssl-1.0.2.tar.gz"
+  url "https://www.openssl.org/source/openssl-1.0.2a.tar.gz"
+  mirror "https://raw.githubusercontent.com/DomT4/LibreMirror/master/OpenSSL/openssl-1.0.2a.tar.gz"
   # Use sha1 for Tiger (which needs OpenSSL to compute sha256 hashes)
-  sha1 "2f264f7f6bb973af444cd9fc6ee65c8588f610cc"
+  sha1 "46ecd325b8e587fa491f6bb02ad4a9fb9f382f5f"
```
Comment (DomT4) [inline Library/Formula/openssl.rb]: Making this SHA256 again would be nice. See the comment I just referenced for reasoning.

**Comment by DomT4** [comment]
The OpenSSL website is currently dying under the load, hence bot failure. I'll grab the tarball ASAP and upload it to the mirror.

File: Library/Formula/openssl.rb (hunk 1 above)
Comment (vszakats) [inline Library/Formula/openssl.rb]: Rolled back from SHA256 after reading the comment. I wasn't aware of that discussion, thanks for the pointer, SHA256 it is now.

**Comment by vszakats** [comment]
I've uploaded it here, if it helps: https://transfer.sh/soTJr/openssl-1.0.2a.tar.gz

**Comment by DomT4** [comment]
Already grabbed them :). Just waiting for my awful upload speed to finish pushing them to Github. Should be up in the next couple minutes.

**Comment by henrikhodne** [comment]
Any reason why the SHA-sum was switched to SHA-256? OpenSSL only gives MD5 and SHA-1 on their website, so SHA-256 will be harder to verify against OpenSSL checksums.

**Comment by mikemcquaid** [comment]
@henrikhodne We're using `sha256` globally in Homebrew now as it's a more robust checksum.

**Comment by vszakats** [comment]
Left an Issue about the problem on OpenSSL's web project: https://github.com/openssl/web/issues/1

**Comment by DomT4** [comment]
> OpenSSL only gives MD5 and SHA-1 on their website

I've filed several requests upstream to start publishing the SHA256SUM. It's something they've talked about in the past as well, so I'm not quite sure why they aren't yet. MD5 is just painful.

**Comment by mikemcquaid** [comment]
Might be worth submitting a PR to that web project as well as an issue.

**Comment by DomT4** [comment]
Kicked *another* email about this upstream as well.

**Comment by DomT4** [comment]
@mikemcquaid Unrelated failure here. Looks like one of our bottles has broken:

```
==> Downloading https://homebrew.bintray.com/bottles/cfengine-3.6.4.mavericks.bottle.tar.gz
Downloaded to: /Library/Caches/Homebrew/cfengine-3.6.4.mavericks.bottle.tar.gz
SHA1: daef9be4a37f5a78d68ab55f55da225a095b3d55
SHA256: 1df65a7c0c4be50e4815eac202d416841e4788fc531d13d8f2d63e165df6332c
==> Retrying download
==> Downloading https://homebrew.bintray.com/bottles/cfengine-3.6.4.mavericks.bottle.tar.gz
Warning: Formula reports different SHA1: 784285e682e6b170bef1d36d8c60112a4b90bad5
Downloaded to: /Library/Caches/Homebrew/cfengine-3.6.4.mavericks.bottle.tar.gz
SHA1: daef9be4a37f5a78d68ab55f55da225a095b3d55
SHA256: 1df65a7c0c4be50e4815eac202d416841e4788fc531d13d8f2d63e165df6332c
```

Everything else worked fine.

**Comment by mikemcquaid** [comment]
Thanks @vszakats and @DomT4!

**Comment by tdsmith** [comment]
Say, does "1.0.2a" get sorted as less than "1.0.2" because "alpha"? I don't see this in `outdated` and I don't get it on `upgrade`.

**Comment by DomT4** [comment]
> Say, does "1.0.2a" get sorted as less than "1.0.2" because "alpha"? I don't see this in outdated and I don't get it on upgrade.

I think this is a persistent problem with the openssl version lettering. There have been mentions of it before. Not sure what the best fix is really :cry:

**Comment by mikemcquaid** [comment]
@tdsmith Yup, sucks.

**Comment by DomT4** [comment]
Maybe we could insert custom versoning. ` version "1.0.2_1" ` or something. Sucks to deviate from upstream though. Maybe some custom OpenSSL logic in outdated and/or upgrade to be more sensitive to the lettering. Meh. Not great either way.

**Comment by mikemcquaid** [comment]
Should probably work something out. Hmm.

**Comment by DomT4** [comment]
I'll go take an axe to various things tonight and see if I can hack something together. If anyone wants to beat me to any ideas, more than welcome to do so :smile_cat:.

**Comment by mikemcquaid** [comment]
I'm on it.

**Comment by mikemcquaid** [comment]
https://github.com/Homebrew/homebrew/pull/37864

**Comment by DomT4** [comment]
Awesome! Thanks. If it works, I'll port it over to versions' OpenSSL.

**Pull request closed without merging**